examples/signals/signals.html

```diff
@@ -0,0 +1,120 @@
+ following copyright and licenses apply:
+
+ Copyright 2020 RDK Management
+
+ Licensed under the Apache License, Version 2.0 (the License);
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" />
+    <script src="../../../devtools/lightning-inspect.js"></script>
+</head>
+<body style="margin: 0; padding: 0">
+    <script type="module">
+        import lng from '../../../src/lightning.mjs';
+        //attachInspector(lng)
+
+        window.onload = function() {
+            class SubBox extends lng.Component {
+                static _template() {
+                    return {
+                        w: 50,
+                        h: 50,
+                        rect: true,
+                        color: 0xff33ffff,
+                    }
+                }
+
+                _init() {
+                    this.signal('passSignal1')
+                    this.signal('passSignal2')
+                }
+            }
+
+            class Box extends lng.Component {
+                static _template() {
+                    return {
+                        w: 300,
+                        h: 300,
+                        rect: true,
+                        color: 0xffffff00,
+                        SubBox: {
+                            type: SubBox,
+                            passSignals: {
+                                passSignal1: true,
+                                passSignal2: 'renamedPassSignal2'
+                            }
+                        }
+                    }
+                }
+
+                _init() {
+                    this.signal('signalHandler1')
+                    this.signal('signalHandler2')
+                    this.signal('signalHandler3')
+                    this.signal('signalHandler4')
+                }
+            }
+
+            class BasicUsageExample extends lng.Application {
+                static _template(ctx) {
+                    return {
+                        w: 900,
+                        h: 900,
+                        Bg: {
+                            src: "../landscape.jpg", scale: 1,
+                        },
+                        Box: {
+                            type: Box,
+                            x: w => w / 2,
+                            y: h => h / 2,
+                            mount: 0.5,
+                            signals: {
+                                signalHandler1: ctx.signalHandler1,
```

Comment:
This introduces a bug.

As the template is `static`, it will be evaluated once you construct the first component and its value will be saved into the class. That parsed template will have closure over the context being passed to `_template(ctx)` meaning that `ctx.signalHandler1` will belong to the first component instance, while new components created from the same class will not have the right context as `_template(` will not be executed again.

I would advise against keeping this in the codebase until it's figured out how to give the proper instance to the template.